packages/core/src/components/TextField/Mask.jsx

```diff
@@ -109,19 +105,24 @@ function toNumber(value) {
  */
 export function maskValue(value = '', mask) {
   if (value && typeof value === 'string') {
+    const hasDigit = value.match(/\d/);
     value = value.trim();
 
-    if (mask === 'currency') {
-      // Format number with commas. If the number includes a decimal,
-      // ensure it includes two decimal points
-      const number = toNumber(value);
-      if (number === undefined) {
-        value = '';
-      } else {
-        value = stringWithFixedDigits(number.toLocaleString('en-US'));
+    if (!hasDigit) {
+      value = '';
+    } else {
+      if (mask === 'currency') {
+        // Format number with commas. If the number includes a decimal,
+        // ensure it includes two decimal points
+        const number = toNumber(value);
+        if (number !== undefined) {
+          value = stringWithFixedDigits(number.toLocaleString('en-US'));
+        }
+      } else if (deliminatedMaskRegex[mask]) {
```

Comment:
Changed `Object.keys(deliminatedMaskRegex).includes(mask)` to `deliminatedMaskRegex[mask]`